Streamline documentation to reflect project "piper" (#875)
SAP/jenkins-library: documentation/docs/configuration.md

> Configuration is done via a yml-file, located at `.pipeline/config.yml` in the **master branch** of your source code repository.

Reviewer: ```suggestion
Configure your project through a yml-file, which is located at `.pipeline/config.yml` in the **master branch** of your source code repository.
```

> !!! note "Cloud SDK Pipeline"

Reviewer 2: I think it is sad to have such kind of different approaches (`pipeline_config.yml`, `.pipeline/config.yml`). Having such kind of notes here is a smell. The dotfile/dotfolder approach is a commonly used standard.

Author: I agree, the current state is not good. It hurt writing that down, but as far as I can tell modifying it now while keeping backwards compatibility is not totally trivial.

> !!! note "Cloud SDK Pipeline"

Reviewer: ```suggestion
**Note:**
```

>     Configuration for Cloud SDK Pipeline is done in a file called `pipeline_config.yml`. It is documented [here](https://github.com/SAP/cloud-s4-sdk-pipeline/blob/master/configuration.md).

Reviewer: ```suggestion
    Cloud SDK Pipelines are configured in a file called `pipeline_config.yml`. See [SAP Cloud SDK Pipeline Configuration Docs](https://github.com/SAP/cloud-s4-sdk-pipeline/blob/master/configuration.md).
```
